src/adt.rs

```diff
@@ -140,4 +322,151 @@ pub(crate) mod tests {
             assert_eq!(res, values);
         }
     }
+
+    mod memory_tests {
```

Comment:
what are you doing here? You seem to be testing an object that is not used...

Reply:
separate vector adt tests from memory adt local tests